js/ui/html_editor/ui.html_editor.js

```diff
@@ -509,22 +509,39 @@ const HtmlEditor = Editor.inherit({
     },
 
     _optionChanged: function(args) {
+        const needCallBase = this._isEditorUpdating;
+        let callBaseArgs = args;
+
         switch(args.name) {
             case 'value':
                 if(this._quillInstance) {
                     if(this._isEditorUpdating) {
                         this._isEditorUpdating = false;
+                        if('_previousValue' in this) {
+                            callBaseArgs = {
+                                ...args,
+                                previousValue: this._previousValue
+                            };
+                            this._valueChangeCalled = true;
+                        }
                     } else {
                         const updatedValue = this._isMarkdownValue() ? this._updateValueByType('HTML', args.value) : args.value;
+
+                        this._previousValue = args.previousValue;
                         this._updateHtmlContent(updatedValue);
+                        delete this._previousValue;
                     }
                 } else {
                     this._$htmlContainer.html(args.value);
                 }
 
                 this._setSubmitValue(args.value);
 
-                this.callBase(args);
+                if(Boolean(needCallBase) === false && this._valueChangeCalled) {
+                    this._valueChangeCalled = false;
+                } else {
+                    this.callBase(callBaseArgs);
+                }
```

Comment:
oh, cannot we simplify the solution? 2 private caches in instance + 2 new variables seems to be too much for such a problem.

Maybe we can write smth like this
```javascript
                        this._suppressValueChangeAction();
                        this._updateHtmlContent(updatedValue);
                        this._resumeValueChangeAction();
                        ...
                        
                        
                // NOTE: value can be optimized by Quill
                const value = this.option('value');
                if(value !== args.previousValue) {
                    this._setSubmitValue(value);
                    this.callBase({ ...args, value });
                }
```